graphql_compiler/compiler/sqlalchemy_extensions.py

```diff
@@ -75,4 +102,18 @@ def visit_bindparam(self, bindparam, **kwargs):
             bindparam.expanding = False
             return super(BindparamCompiler, self).visit_bindparam(bindparam, **kwargs)
 
-    return str(BindparamCompiler(dialect, query).process(query))
+    return str(BindparamCompiler(printing_dialect, query).process(query))
+
+
+def bind_parameters_to_query_string(query: str, parameters: Dict[str, Any]) -> TextClause:
+    """Assign values to query parameters."""
+    bound_parameters = [
+        sqlalchemy.bindparam(
+            parameter_name,
+            value=parameter_value,
+            expanding=isinstance(parameter_value, (list, tuple)),
+        )
+        for parameter_name, parameter_value in parameters.items()
+    ]
+
+    return sqlalchemy.text(query).bindparams(*bound_parameters)
```

Comment:
This function literally parses the query with a regex looking for ":<key>" and then matches the parameters to the keys found. I don't fully understand it, but I hope it's just a validation step, and not a find-and-replace strategy.